src/modules/reachRandomJointPositions/src/module.cpp

```diff
@@ -238,129 +254,163 @@ bool reachRandomJointPositionsModule::configure(ResourceFinder &rf)
+            listOfDesiredPositions.push_back(desiredPositions(center,return_point_waiting_period,
+                                                              true,desiredPositions::ROW_OUT));
+
+            //Print list of desired positions
+            std::cout  << "List of desired positions: " << std::endl;
+            for(int i=0; i<int(listOfDesiredPositions.size()); i++)
+            {
+                std::cout << listOfDesiredPositions[i].toString() << std::endl;
+            }
+
+            isTheRobotInReturnPoint.open("/"+moduleName+"/"+controlledJoints[0].part_name+"/isTheRobotInReturnPoint:o");
+            useFurtherPosForFitting.open("/"+moduleName+"/"+controlledJoints[0].part_name+"/useFurtherPosForFitting:o");
+            
+            break;
         }
-        for(int j=0; j < (int)semi_nr_of_lines[1]; j++ )
-        {
-            //Draw upper row
-            yarp::sig::Vector row_center(2), row_lower(2), row_upper(2);
-            row_upper[1] = row_lower[1] = row_center[1] = center[1]+j*controlledJoints[1].delta;
-            row_center[0] = center[0];
-            row_lower[0] = controlledJoints[0].lower_limit;
-            row_upper[0] = controlledJoints[0].upper_limit;
-            listOfDesiredPositions.push_back(desiredPositions(row_center,static_pose_period));
-            listOfDesiredPositions.push_back(desiredPositions(row_lower,static_pose_period));
-            listOfDesiredPositions.push_back(desiredPositions(row_upper,static_pose_period));
-            listOfDesiredPositions.push_back(desiredPositions(row_center,static_pose_period));
-            listOfDesiredPositions.push_back(desiredPositions(center,return_point_waiting_period,true));
-
-            //Draw lower row
-            row_upper[1] = row_lower[1] = row_center[1] = center[1]-j*controlledJoints[1].delta;
-            row_center[0] = center[0];
-            row_lower[0] = controlledJoints[0].lower_limit;
-            row_upper[0] = controlledJoints[0].upper_limit;
-            listOfDesiredPositions.push_back(desiredPositions(row_center,static_pose_period));
-            listOfDesiredPositions.push_back(desiredPositions(row_lower,static_pose_period));
-            listOfDesiredPositions.push_back(desiredPositions(row_upper,static_pose_period));
-            listOfDesiredPositions.push_back(desiredPositions(row_center,static_pose_period));
-            listOfDesiredPositions.push_back(desiredPositions(center,return_point_waiting_period,true));
-        }
+
+        case RANDOM:
+        case GRID_VISIT:
+            break;
+
+        default:
+            break;
+    }
+
+    //Latch the remote timestamp for synchronising local one
+    this->latchTimestampSync();
+
+    //Wait for "isTheRobotInReturnPoint:o" and "useFurtherPosForFitting:o" to be
+    //connected to another port
+    if( !returnFlagPort.empty() )
+    {
+        while(!Network::isConnected(isTheRobotInReturnPoint.getName(), returnFlagPort)) {};
```

Comment:
It better to add some sort of the delay to prevent stressing to much the network?